MatrixSDK/Crypto/MXCrypto.m

```diff
@@ -817,6 +818,24 @@ - (void)handleDeviceListsChanges:(MXDeviceListResponse*)deviceLists
 #endif
 }
 
+- (void)handleToDeviceEvent:(MXEvent *)event onComplete:(void (^)(void))onComplete
+{
+    switch (event.eventType)
+    {
+        case MXEventTypeRoomKey:
+        {
+            [self handleRoomKeyEvent:event onComplete:onComplete];
+            break;
+        }
+            
+        default:
+            onComplete();
+            break;
+    }
+    [self.mxSession.eventStreamService dispatchOnLiveToDeviceWithEvent:event];
```

Comment:
Do we really need to have this dependency on MXSession here?
Can this block stay in  MXSession [here](https://github.com/matrix-org/matrix-ios-sdk/pull/1386/files#diff-480363b404ebe0778edd253e1f353db5b0476b367fc5f986440f7814ec79912eL1897)?

I cannot see this method used outside the SDK. The related app [PR](https://github.com/vector-im/element-ios/pull/5743) do not use it.

Reply:
`mxSession` is already a hard dependency in crypto, i just re-used it.
Android handles `toDeviceEvents` in crypto like this and it made the caller site clearer in some instances as we were passing a `toDevice` variable.
But I really don't feel strong about it, so happy to move it back, Will create a new PR.